ObservatoryExplorer/Explorer.cs

```diff
@@ -109,12 +110,24 @@ public Scan ConvertBarycentre(ScanBaryCentre barycentre, Scan childScan)
 
             return barycentreScan;
         }
+        public void SetSystem(string potentialNewSystem)
+        {
+            if (string.IsNullOrEmpty(currentSystem) || currentSystem != potentialNewSystem)
+            {
+                currentSystem = potentialNewSystem;
+                if (ExplorerWorker.settings.OnlyShowCurrentSystem && !ObservatoryCore.IsLogMonitorBatchReading)
+                {
+                    ObservatoryCore.ClearGrid(ExplorerWorker, new ExplorerUIResults());
+                    Clear();
+                }
+            }
+        }
 
         public void ProcessScan(Scan scanEvent, bool readAll)
         {
             if (!readAll)
             {
-                string criteriaFilePath = ((ExplorerSettings)ExplorerWorker.Settings).CustomCriteriaFile;
+                string criteriaFilePath = ExplorerWorker.settings.CustomCriteriaFile;
```

Comment:
Ah, making the actual typed object internal is a nice change for code clarity here.